docs/en/building/basic-usage.md

```diff
@@ -0,0 +1,118 @@
+# Basic Usage
+
+## Creating a Dojo application
+
+The first step for building a Dojo application is to install the CLI
+
+> Command line
+
+```bash
+npm install -g @dojo/cli @dojo/cli-create-app
+```
+
+This command installs the Dojo command-line tool (`@dojo/cli`) and the Dojo CLI module to create an application (`@dojo/cli-create-app`). There are a number of CLI modules that add commands and functionality to the dojo command-line.
+
+Now create an application using the dojo command.
+
+> Command line
+
+```bash
+dojo create app --name my-first-app
+```
+
+This command will generate a basic project structure in the `my-first-app` directory. This project will include a `package.json` file, an example Dojo application, and all of the project dependencies will be pre-installed. It will include two additional Dojo CLI modules. [@dojo/cli-build-app](https://github.com/dojo/cli-build-app) and [@dojo/cli-test-intern](https://github.com/dojo/cli-test-intern).
+
+## Building
+
+There are several types of build targets or `mode`s that Dojo's CLI tool supports. All of these modes can be seen in the various `package.json` scripts generated by `dojo create app`.
+
+Run the following to create an optimized build ready for production.
+
+> Command line
+
+```bash
+npm run build
+```
+
+This build will use the `dist` mode to create an application bundle and output the results in the `output/dist` directory.
+
+## Serving and Watching
+
+A web server can be started with the `--serve` flag while running in `dev` or `dist` modes. By default the application is served on port 9999. This can be cahnged with the `--port` flag. Dojo's build tool can also watch the application for changes and automatically rebuild with the `--watch` flag.
+
+The generated `package.json` includes the `dev` script which uses these flags to serve a built version of the application and automatically rebuild it when files change on disk.
+
+> Command line
+
+```bash
+npm run dev
+```
+
+The application will also have source maps available. This allows the debugger to map the built JavaScript code back to the original TypeScript code that lives in the `src/` directory.
+
+For more information on how to configure the build options see the [@dojo/cli-build-app](https://github.com/dojo/cli-build-app/blob/master/README.md#configuration) README.
+
+## Testing
```

Comment:
I don't think we need to talk about testing in detail for the build reference guide over explaining how to build the `unit` and `functional` bundles (with a link out to the testing reference guide, when it's done!)

Reply:
I can cut this down. The purpose was to acknowledge that there are unique build targets for functional and unit tests and perhaps provide a little insight on why this is the case.